drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -31,6 +31,96 @@
 #define z_apb4_prescaler(v) LL_RCC_APB4_DIV_ ## v
 #define apb4_prescaler(v) z_apb4_prescaler(v)
 
+/* Macro to check for clock feasability */
+/* It is Cortex M7's responsibility to setup clock tree */
+/* This check should only be performed for the M7 core code */
+#ifdef CONFIG_CPU_CORTEX_M7
+
+/* Define primary oscillator frequencies */
+/* Suppress the cast to uint32_t which */
+/* prevents from compare with #if > operator*/
+/* original defines in stm32h7xx_hal_conf.h*/
+#define HSI_FREQ 64000000UL	/* HSI_VALUE ((uint32_t)64000000) */
+/* HSE_VALUE overridden by the build system without C cast to uint*/
+/* Build system doesn't provide the UL type suffix for HSE_VALUE */
+/* Force HSE_FREQ to be cast to preprocessor UL to prevent overflow*/
+/* Only use these constants in preprocessor expressions */
+#define HSE_FREQ HSE_VALUE
+#define CSI_FREQ 4000000UL	/* CSI_VALUE ((uint32_t)4000000) */
+
+/* Choose PLL SRC */
+#ifdef CONFIG_CLOCK_STM32_PLL_SRC_HSI
+#define PLLSRC_FREQ  HSI_FREQ
```

Comment:
`HSI_FREQ` can be divided by `HSIDIV`. I have it such in one of my projects. In such case the checks for `SYSCLK_FREQ > SYSCLK_FREQ_MAX` fails by mistake.

Reply:
Now I see you probably do not have there support for `HSIDIV != 1` at all. I found that also have it hardcoded in my project...

Reply:
Yes you are right, HSI can be divided.

What symbol to you use to provide the HSIDIV value ? So I can add it to the driver.

Reply:
We probably need to include a new value to Kconfig for HSIDIV. I had there just hardcoded line setting that divider to 8, so for me the HSI freq was always 8MHz. When I was trying your PR I just hit your guards, that are right for `HSIDIV == 1`.

Reply:
To add the HSI divisor configuration I suggest: 
Name:
1. CLOCK_STM32_HSI_DIVISOR : 
2. Parameters: default to 1, range: 1 8, allowed values: 1 2 4 8

Reply:
No problem for me. Please note that I'm not a Zephyr developer/maintainer. I just use it on STM32H7 platform so your patches are handy for me - thank you. I had also my changes on similar topic, but never had time to polish them and make public.

Reply:
Check out the latest commit, I have tested this implementation with:
```
# HSI divided to 8MHz test
CONFIG_CLOCK_STM32_SYSCLK_SRC_HSI=y
CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=8000000
CONFIG_CLOCK_STM32_HSI_DIVISOR=8
```
and also from PLL at 480MHz:
```
# HSI divided to 8MHz + PLL 480MHz test
CONFIG_CLOCK_STM32_PLL_SRC_HSI=y
CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=480000000
CONFIG_CLOCK_STM32_HSI_DIVISOR=8
CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL=y
CONFIG_CLOCK_STM32_PLL_M_DIVISOR=1
CONFIG_CLOCK_STM32_PLL_N_MULTIPLIER=120
CONFIG_CLOCK_STM32_PLL_P_DIVISOR=2
CONFIG_CLOCK_STM32_PLL_Q_DIVISOR=2
CONFIG_CLOCK_STM32_PLL_R_DIVISOR=2
CONFIG_CLOCK_STM32_D1CPRE=1
CONFIG_CLOCK_STM32_HPRE=2
CONFIG_CLOCK_STM32_D2PPRE1=2
CONFIG_CLOCK_STM32_D2PPRE2=2
CONFIG_CLOCK_STM32_D1PPRE=2
CONFIG_CLOCK_STM32_D3PPRE=2
```

Reply:
I have tested it on my board. Works fine.

Reply:
This addition will be added to PR description.

The code will have to be reviewed by approved reviewers to see if it can be embedded in this PR.
@ABOSTM I would be glad to here what you think 👍

Reply:
> I have tested it on my board. Works fine.

Please name the board and/or reference to it if it is already available in Zephyr's main repo.

Reply:
> To add the HSI divisor configuration I suggest: 
> Name:
> 1. CLOCK_STM32_HSI_DIVISOR : 
> 2. Parameters: default to 1, range: 1 8, allowed values: 1 2 4 8

I think DIV instead of DIVISOR will be better, cause shorter.

Reply:
> > I have tested it on my board. Works fine.
> 
> Please name the board and/or reference to it if it is already available in Zephyr's main repo.

It's a custom board, not suitable for main repo unfortunately.

Reply:
> No problem for me. Please note that I'm not a Zephyr developer/maintainer. I just use it on STM32H7 platform so your patches are handy for me - thank you. I had also my changes on similar topic, but never had time to polish them and make public.
> BTW so far all your patches (including ETH) work on my custom board with STM32H750VBT6.

Nice to hear, you can use it in your project. I hope, you can profit from the project more and more with it's growth.

Reply:
`CLOCK_STM32_HSI_DIVISOR` I maybe left the DIVISOR suffix here, to be in same manner as other existing values like `CLOCK_STM32_PLL_M_DIVISOR`.

Reply:
> I think DIV instead of DIVISOR will be better, cause shorter.
I agree.
But I chose DIVISOR to align with the PLL symbols:
```
CLOCK_STM32_PLL_M_DIVISOR
CLOCK_STM32_PLL_R_DIVISOR
CLOCK_STM32_PLL_Q_DIVISOR
```